### PR TITLE
Python sdk anyof allof root model

### PR DESCRIPTION
**Pull request by padznich**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package || exit
  ./bin/generate-samples.sh ./bin/configs/*.yaml || exit
  ./bin/utils/export_docs_generators.sh || exit
  ``` 
  (For Windows users, please run the script in [WSL](https://learn.microsoft.com/en-us/windows/wsl/install))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.

---
## Summary by cubic
Moves Python anyOf/oneOf models to Pydantic v2 RootModel and updates generated models, docs, and samples to Pydantic’s validation/serialization APIs. Fixes RootModel edge cases (e.g., Pig) and strips directory traversal from download filenames; tests and samples updated.

- **New Features**
  - anyOf/oneOf now use RootModel[Union[...]] with the value on .root; attribute access delegates to .root.
  - Inline enums (including lists) use Literal[...] with Field alias/description.
  - Docs/samples now use model_validate_json/model_dump_json/model_dump/model_validate.

- **Migration**
  - Use model_validate_json/model_dump_json/model_dump/model_validate instead of to_json/from_json/to_dict/to_str; legacy helpers removed from generated models.
  - For anyOf/oneOf, access the underlying value via .root.

<sup>Written for commit e260bb207214dd5f40eca8ba9f888675986ecd18. Summary will update on new commits.</sup>

### CODE REVIEW

**Comment by wing328** [comment]
looks like your PR is not based on the latest master

can you please file a new one based on the latest master or merge the latest master into your branch (and update both samples and docs)?

**Comment by wing328** [comment]
cc @cbornet (2017/09) @tomplus (2018/10) @krjakbrjak (2023/02) @fa0311 (2023/10) @multani (2023/10)

**Comment by wing328** [comment]
i've filed https://github.com/OpenAPITools/openapi-generator/pull/22963 with updated samples but some tests failed.

can you please review those CI test failure?

File: modules/openapi-generator/src/main/resources/python/model_generic.mustache [hunk 1]
```diff
@@ -121,278 +156,28 @@ class {{classname}}({{#parent}}{{{.}}}{{/parent}}{{^parent}}BaseModel{{/parent}}
...
-            return None
+    def from_json(cls, json_str: str) -> Self:
+        """Returns the object represented by the json string"""
+        return cls.model_validate_json(json_str)
 
-        if not isinstance(obj, dict):
-            return cls.model_validate(obj)
+    def to_json(self) -> str:
+        """Returns the JSON representation of the actual instance"""
+        return json.dumps(self.model_dump(by_alias=True, exclude_none=True))
```
[the remark this answers is not in the document]
Comment (padznich) [reply modules/openapi-generator/src/main/resources/python/model_generic.mustache]: It changes the output
FROM:
```
{"id":1,"name":"test"}
```

TO:
```
{"id": 1, "name": "test"}
```
